tests/unit/pipeline/value_providers/test_regex_value_provider.py

```diff
@@ -11,7 +11,7 @@
 def subject_with_named_groups():
     return RegexValueProvider(
         regex="(?P<name>[a-z]+) (?P<age>[0-9]+)",
-        data=StubbedValueProvider(["john 42", "jane 32"]),
+        data=StubbedValueProvider(["I am john 42", "I am jane 32"]),
```

Comment:
Should we keep one of these the same so we can test that it matches if its the whole string?